diff --git a/BUILD.gn b/BUILD.gn
index 92277c2e52fccd844f0eca42515f1a27f58f7893..9ebfd30d4faddd1bda60cad3e0bd690c3b291ad5 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -208,9 +208,15 @@ config("internal_config") {
     "$target_gen_dir",
   ]
 
+  if (is_win) {
+    cflags = [
+      "/wd4275",  # dll export
+    ]
+  }
+
   defines = []
 
-  if (is_component_build) {
+  if (true) {
     defines += [ "BUILDING_V8_SHARED" ]
   }
 }
diff --git a/src/profiler/profiler-listener.cc b/src/profiler/profiler-listener.cc
index 51327685059037f2b0614feaafc75aef81512bca..67f6f30275ee8bfd7ca463830b0c13b5c75bb2d7 100644
--- a/src/profiler/profiler-listener.cc
+++ b/src/profiler/profiler-listener.cc
@@ -98,7 +98,7 @@ void ProfilerListener::CodeCreateEvent(CodeEventListener::LogEventsAndTags tag,
         continue;
       int position = it.source_position().ScriptOffset();
       int line_number = script->GetLineNumber(position) + 1;
-      line_table->SetPosition(it.code_offset(), line_number);
+      line_table->SetPosition(it.code_offset(), line_number > 0 ? line_number : line);
     }
   }
   rec->entry =
