diff --git a/BUILD.gn b/BUILD.gn
index 8895103222b4ee8ebf212d5c3e1e030cf62a4fd5..764cd0d3b5546a3c744923a1f3cc41ea32027af2 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -134,7 +134,13 @@ config("internal_config") {
 
   include_dirs = [ "." ]
 
-  if (is_component_build) {
+  if (is_win) {
+    cflags = [
+      "/wd4275",  # dll export
+    ]
+  }
+
+  if (true) {
     defines = [ "BUILDING_V8_SHARED" ]
   }
 }
