diff --git a/BUILD.gn b/BUILD.gn
index 49707659723ca26ce786c6b04e20fac78bed2405..e20529b98a73a6e8087a51acfe43774fde156b10 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -179,7 +179,13 @@ config("internal_config") {
 
   include_dirs = [ "." ]
 
-  if (is_component_build) {
+  if (is_win) {
+    cflags = [
+      "/wd4275",  # dll export
+    ]
+  }
+
+  if (true) {
     defines = [ "BUILDING_V8_SHARED" ]
   }
 }
diff --git a/src/profiler/profiler-listener.cc b/src/profiler/profiler-listener.cc
index bd2f158e6072386ba8e16c5aa50482d9379b663f..657b99f1e05c204df518e54d0a2f7296e3858368 100644
--- a/src/profiler/profiler-listener.cc
+++ b/src/profiler/profiler-listener.cc
@@ -96,7 +96,7 @@ void ProfilerListener::CodeCreateEvent(CodeEventListener::LogEventsAndTags tag,
       int position = it.source_position().ScriptOffset();
       int line_number = script->GetLineNumber(position) + 1;
       int pc_offset = it.code_offset() + offset;
-      line_table->SetPosition(pc_offset, line_number);
+      line_table->SetPosition(pc_offset, line_number > 0 ? line_number : line);
     }
   }
   rec->entry = NewCodeEntry(
