diff --git a/BUILD.gn b/BUILD.gn
index 456a318c1c60527ae264cc3b8326cbd860dc5d2f..27291944c55c2b8ac343344419f759f8dbe2e55f 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -195,7 +195,13 @@ config("internal_config") {
     "$target_gen_dir",
   ]
 
-  if (is_component_build) {
+  if (is_win) {
+    cflags = [
+      "/wd4275",  # dll export
+    ]
+  }
+
+  if (true) {
     defines = [ "BUILDING_V8_SHARED" ]
   }
 }
diff --git a/src/profiler/profiler-listener.cc b/src/profiler/profiler-listener.cc
index 70d907b21659702e2543c6a18f7b855d112d786f..8a1d417e1eaa7a7d831a86403ceb37792a10dde6 100644
--- a/src/profiler/profiler-listener.cc
+++ b/src/profiler/profiler-listener.cc
@@ -98,7 +98,7 @@ void ProfilerListener::CodeCreateEvent(CodeEventListener::LogEventsAndTags tag,
         continue;
       int position = it.source_position().ScriptOffset();
       int line_number = script->GetLineNumber(position) + 1;
-      line_table->SetPosition(it.code_offset(), line_number);
+      line_table->SetPosition(it.code_offset(), line_number > 0 ? line_number : line);
     }
   }
   rec->entry =
