diff --git a/BUILD.gn b/BUILD.gn
index 8492cb5f624156ab7b7bc9cd6c06ad996eb45182..9ec52a4ef6c13b89a4bd5f364e0447eb889bc1bd 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -175,7 +175,13 @@ config("internal_config") {
 
   include_dirs = [ "." ]
 
-  if (is_component_build) {
+  if (is_win) {
+    cflags = [
+      "/wd4275",  # dll export
+    ]
+  }
+
+  if (true) {
     defines = [ "BUILDING_V8_SHARED" ]
   }
 }
diff --git a/src/profiler/profiler-listener.cc b/src/profiler/profiler-listener.cc
index fecfdb66b099388cfeb71c3c7b4ef8832bfac2a6..392b34a3037b4690cb990f72216da4935bec97f2 100644
--- a/src/profiler/profiler-listener.cc
+++ b/src/profiler/profiler-listener.cc
@@ -96,7 +96,7 @@ void ProfilerListener::CodeCreateEvent(CodeEventListener::LogEventsAndTags tag,
       int position = it.source_position().ScriptOffset();
       int line_number = script->GetLineNumber(position) + 1;
       int pc_offset = it.code_offset() + offset;
-      line_table->SetPosition(pc_offset, line_number);
+      line_table->SetPosition(pc_offset, line_number > 0 ? line_number : line);
     }
   }
   rec->entry = NewCodeEntry(
