diff --git a/BUILD.gn b/BUILD.gn
index 8587356ddcca34cdcad716ba7395666ec7ce4380..a75c13885a8502419401d62c684ae96264e146ac 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -151,9 +151,9 @@ config("libsampler_config") {
 # This config should only be applied to code using V8 and not any V8 code
 # itself.
 config("external_config") {
-  if (is_component_build) {
+  # if (is_component_build) {
     defines = [ "USING_V8_SHARED" ]
-  }
+  # }
   include_dirs = [ "include" ]
   if (v8_enable_inspector_override) {
     include_dirs += [ "$target_gen_dir/include" ]
@@ -610,6 +610,7 @@ if (v8_use_external_startup_data) {
       ":js2c_extras",
     ]
 
+    public_deps = [ "//electron/build/node:build_node" ]
     # NOSORT
     sources = [
       "$target_gen_dir/libraries.bin",
@@ -619,7 +620,7 @@ if (v8_use_external_startup_data) {
     ]
 
     outputs = [
-      "$root_out_dir/natives_blob.bin",
+      "$root_out_dir/muon_natives_blob.bin",
     ]
 
     script = "tools/concatenate-files.py"
@@ -690,7 +691,8 @@ action("run_mksnapshot") {
   }
 
   if (v8_use_external_startup_data) {
-    outputs += [ "$root_out_dir/snapshot_blob.bin" ]
+    public_deps = [ "//electron/build/node:build_node" ]
+    outputs += [ "$root_out_dir/muon_snapshot_blob.bin" ]
     args += [
       "--startup_blob",
       rebase_path("$root_out_dir/snapshot_blob.bin", root_build_dir),
diff --git a/gni/v8.gni b/gni/v8.gni
index 3759572b935157e65237443c84ef456c508c2010..8b8f80acc88bba44f205daf63f3968fc477f91bf 100644
--- a/gni/v8.gni
+++ b/gni/v8.gni
@@ -83,7 +83,7 @@ if (is_posix && v8_enable_backtrace) {
 
 # All templates should be kept in sync.
 template("v8_source_set") {
-  source_set(target_name) {
+  loadable_module(target_name) {
     forward_variables_from(invoker, "*", [ "configs" ])
     configs += invoker.configs
     configs -= v8_remove_configs
