diff --git a/BUILD.gn b/BUILD.gn
index 4e9364b7eabd9bf024632f7df41ae66a436fc632..51630abd9d7f2766042518f35db7f662e0c26f91 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -3,6 +3,7 @@
 # found in the LICENSE file.
 
 import("//third_party/icu/config.gni")
+import("//build/config/chrome_build.gni")
 
 if (is_android) {
   import("//build/config/android/rules.gni")
@@ -12,6 +13,43 @@ if (is_mac) {
   import("//build/config/sanitizers/sanitizers.gni")
 }
 
+icui18n_name = "icui18n"
+icuuc_name = "icuuc"
+if (muon_build) {
+  group("icui18n") {
+    public_deps = [ "//electron/build/node" ]
+    public_configs = [ ":icu_config" ]
+  }
+
+  group("icuuc") {
+    public_deps = [ "//electron/build/node" ]
+    public_configs = [ ":icu_config" ]
+  }
+
+  group("icudata") {
+    public_deps = [ "//electron/build/node:build_node" ]
+  }
+
+  config("icu_config_chrome") {
+    if (!is_component_build) {
+      defines = [ "U_STATIC_IMPLEMENTATION" ]
+    }
+  }
+
+  group("icu_chrome") {
+    public_deps = [
+      ":icui18n_chrome",
+      ":icuuc_chrome",
+    ]
+
+    public_configs = [
+      ":icu_config_chrome"
+    ]
+  }
+  icui18n_name = "icui18n_chrome"
+  icuuc_name = "icuuc_chrome"
+}
+
 # Meta target that includes both icuuc and icui18n. Most targets want both.
 # You can depend on the individually if you need to.
 group("icu") {
@@ -33,9 +71,11 @@ config("icu_config") {
     "U_ENABLE_DYLOAD=0",
   ]
 
+  if (!muon_build) {
   if (!is_component_build) {
     defines += [ "U_STATIC_IMPLEMENTATION" ]
   }
+  }
 
   include_dirs = [
     "source/common",
@@ -100,7 +140,7 @@ config("icu_code") {
   }
 }
 
-component("icui18n") {
+component(icui18n_name) {
   # find  source/i18n -maxdepth 1  ! -type d  | egrep  '\.(c|cpp|h)$' |\
   # sort | sed 's/^\(.*\)$/    "\1",/'
   sources = [
@@ -525,7 +565,7 @@ component("icui18n") {
 
   defines = [ "U_I18N_IMPLEMENTATION" ]
   deps = [
-    ":icuuc",
+    ":$icuuc_name",
   ]
 
   # ICU uses RTTI, replace the default "no rtti" config.
@@ -542,7 +582,7 @@ component("icui18n") {
   public_configs = [ ":icu_config" ]
 }
 
-component("icuuc") {
+component(icuuc_name) {
   # find  source/common -maxdepth 1  ! -type d  | egrep  '\.(c|cpp|h)$' |\
   # sort | sed 's/^\(.*\)$/    "\1",/'
   sources = [
@@ -963,6 +1003,7 @@ if (is_android && enable_java_templates) {
   }
 }
 
+if (!muon_build) {
 # TODO(GYP): Gyp has considerations here for QNX and for the host toolchain
 #  that have not been ported over.
 if (is_android) {
@@ -1028,3 +1069,4 @@ if (icu_use_data_file) {
     }
   }
 }
+}
